packages/x-components/src/x-modules/search/components/__tests__/banners-list.spec.ts

```diff
@@ -0,0 +1,131 @@
+ *
+ * @param options - The options to render the component with.
+ * @returns The API for testing the `BannersList` component.
+ */
+function renderBannersList({
+  template = '<BannersList />',
+  banners = getBannersStub(),
+  components
+}: RenderBannersListOptions = {}): RendersBannerListAPI {
+  const localVue = createLocalVue();
+  localVue.use(Vuex);
+  const store = new Store<DeepPartial<RootXStoreState>>({});
+  installNewXPlugin({ store }, localVue);
+
+  XPlugin.resetInstance();
+
+  resetXSearchStateWith(store, { banners });
+  const wrapper = mount(
+    {
+      components: {
+        BannersList,
+        ...components
+      },
+      template
+    },
+    {
+      localVue,
+      store
+    }
+  );
+
+  const bannersListWrapper = wrapper.findComponent(BannersList);
+
+  return {
+    bannersListWrapper,
+    getBanners() {
+      return banners;
+    }
+  };
+}
+
+describe('testing BannersList component', () => {
+  it('is an XComponent', () => {
+    const { bannersListWrapper } = renderBannersList();
+    expect(isXComponent(bannersListWrapper.vm)).toEqual(true);
+  });
+
+  it('has Search as XModule', () => {
+    const { bannersListWrapper } = renderBannersList();
+    expect(getXComponentXModuleName(bannersListWrapper.vm)).toEqual('search');
+  });
+
+  it('renders the banners in the state', () => {
+    const { bannersListWrapper, getBanners } = renderBannersList();
+    const bannersListItems = bannersListWrapper.findAll(getDataTestSelector('banners-list-item'));
+
+    getBanners().forEach((result, index) => {
+      expect(bannersListItems.at(index).text()).toEqual(result.title);
+    });
+  });
+
+  it('does not render any banner if the are none', () => {
+    const { bannersListWrapper } = renderBannersList({ banners: [] });
+    expect(bannersListWrapper.html()).toEqual('');
+  });
+
+  it('allows customizing the banner slot', () => {
+    const { bannersListWrapper } = renderBannersList({
+      template: `
+        <BannersList>
+          <template #banner="{ banner }">
+            <p data-test="banner-slot-overridden">{{ banner.title }}</p>
+          </template>
+        </BannersList>`
+    });
+
+    expect(bannersListWrapper.find(getDataTestSelector('banners-list')).exists()).toBe(true);
+    expect(bannersListWrapper.find(getDataTestSelector('banner-slot-overridden')).exists()).toBe(
+      true
+    );
+  });
+
+  it('allows customizing the default slot', () => {
+    const { bannersListWrapper } = renderBannersList({
+      template: `
+        <BannersList>
+          <template #default="{ banners }">
+            <BaseGrid :items="banners" data-test="default-slot-overridden"/>
+          </template>
+        </BannersList>`,
+      components: { BaseGrid }
+    });
+
+    expect(bannersListWrapper.find(getDataTestSelector('banners-list')).exists()).toBe(false);
+    expect(bannersListWrapper.find(getDataTestSelector('default-slot-overridden')).exists()).toBe(
+      true
+    );
```

Comment:
If you are just going to check that the custom element exists, and that the original doesn't, why bothering with registering another component? For me it just adds noise to the test. You could do the same with a `p` element or any other node, and avoid having to register more components.

Reply:
Just copied the tests from the ResultsList specs. I've changed in both files.